### ISSUE TITLE
operator: don't store defaults in apiserver

### ISSUE THREAD

**Issue by pohly** [intel/pmem-csi#615]
We should use defaulting (aka `Default` in https://pkg.go.dev/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1?tab=doc#JSONSchemaProps, https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190426-crd-defaulting.md). In particular, we should never update the deployment spec with the default values.

That way, defaults can change over time instead of being locked down to some specific default.

This is particularly relevant for the images.

Defaults are currently written to the apiserver when restoring invariant fields. To reproduce:
- `kubectl apply -f operator/examples/pmem-csi.intel.com_v1alpha1_deployment_cr.yaml`
- edit the file and change `pmemPercentage`
- `kubectl apply -f operator/examples/pmem-csi.intel.com_v1alpha1_deployment_cr.yaml`
- `kubectl get -o yaml deployment.pmem-csi.intel.com/example-deployment`:

```
apiVersion: pmem-csi.intel.com/v1alpha1
kind: Deployment
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"pmem-csi.intel.com/v1alpha1","kind":"Deployment","metadata":{"annotations":{},"name":"example-deployment"},"spec":{"controllerResources":{"requests":{"cpu":"200m","memory":"100Mi"}},"nodeResources":{"requests":{"cpu":"200m","memory":"100Mi"}},"pmemPercentage":60}}
  creationTimestamp: "2020-04-24T12:50:24Z"
  generation: 3
  name: example-deployment
  resourceVersion: "35823"
  selfLink: /apis/pmem-csi.intel.com/v1alpha1/deployments/example-deployment
  uid: 403e58fb-f587-49b2-8c7a-dc62fb44f563
spec:
  controllerResources:
    requests:
      cpu: 200m
      memory: 100Mi
  deviceMode: lvm
  driverName: pmem-csi.intel.com
  image: 172.17.42.1:5000/pmem-csi-driver:canary
  imagePullPolicy: IfNotPresent
  logLevel: 5
  nodeRegistrarImage: quay.io/k8scsi/csi-node-driver-registrar:v1.1.0
  nodeResources:
    requests:
      cpu: 200m
      memory: 100Mi
  nodeSelector:
    storage: pmem
  pmemPercentage: 50
  provisionerImage: quay.io/k8scsi/csi-provisioner:v1.2.1
status:
  phase: Running
```

**Comment by avalluri** [comment]
> Defaults are currently written to the apiserver when restoring invariant fields

Yes, this is a bug in code that should be fixed.

**Comment by pohly** [comment]
Resolved in PR #623 by never updating the deployment object.
